src/utils/validation/vault/create.ts

```diff
@@ -0,0 +1,65 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import Big from 'big.js';
+import { TFunction } from 'i18next';
+import * as z from 'zod';
+
+import { displayMonetaryAmount, parseLocaleNumber } from '@/common/utils/utils';
+import { GOVERNANCE_TOKEN_SYMBOL } from '@/config/relay-chains';
+
+type ValidateDepositCollateralParams = {
+  minAmount: MonetaryAmount<CurrencyExt>;
+  balance: MonetaryAmount<CurrencyExt>;
+  governanceBalance: MonetaryAmount<CurrencyExt>;
+  transactionFee: MonetaryAmount<CurrencyExt>;
+};
+
+const validateDepositCollateral = (
+  t: TFunction,
+  params: ValidateDepositCollateralParams
+): z.ZodEffects<z.ZodString, string, string> =>
+  z.string().superRefine((value, ctx) => {
+    const { balance, governanceBalance, minAmount, transactionFee } = params;
+
+    if (governanceBalance.lte(transactionFee)) {
+      return ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: t('insufficient_funds_governance_token', {
+          governanceTokenSymbol: GOVERNANCE_TOKEN_SYMBOL
+        })
+      });
+    }
+
+    if (!value) {
+      return ctx.addIssue({
+        code: z.ZodIssueCode.too_small,
+        minimum: 1,
+        message: t('vault.please_enter_deposit_amount'),
+        inclusive: true,
+        type: 'string'
+      });
+    }
+
+    const parsedValue = parseLocaleNumber(value);
```

Comment:
Minor, but feels a bit messy having two consts in the middle of a load of conditionals :) is there another way to structure this?

Reply:
this will be removed after the TokenInput merge